src/utils/helperFunctions/upload.tsx

```diff
@@ -0,0 +1,174 @@
+import {
+  DispatchAction,
+  FileUpload,
+  UploadField,
+} from "src/screens/UploadFiles";
+import Select from "src/components/formFields/Select";
+import TextInput from "src/components/formFields/TextInput";
+import { ManifestField } from "src/utils/api/manifests";
+
+export const knownFieldNames = [
+  "data_producer_id",
+  "jurisdiction",
+  "sender_id",
+];
+
+export const generateFormData = (state: FileUpload) => {
+  /* eslint-disable @typescript-eslint/no-explicit-any */
+  const formData: any = {
+    data_stream_id: state.datastream,
+    data_stream_route: state.route,
+    version: state.version,
+  };
+
+  state.knownFields.forEach((field: UploadField) => {
+    formData[field.field_name] = field.value;
+  });
+
+  state.extraFields.forEach((field: UploadField) => {
+    formData[field.field_name] = field.value;
+  });
+
+  return formData;
+};
+
+export const isFormValid = (
+  state: FileUpload,
+  dispatch: React.Dispatch<DispatchAction>
+) => {
+  let isValid = true;
+
+  if (!state.file?.name) {
+    dispatch({
+      type: "validateFile",
+      value: "error",
+    });
+    isValid = false;
+  }
+
+  if (!state.datastream || !state.route) isValid = false;
+
+  if (!state.version) {
+    dispatch({
+      type: "validateVersion",
+      value: "error",
+    });
+    isValid = false;
+  }
+
+  for (const field of state.knownFields) {
+    if (field.required && !field.value) {
+      dispatch({
+        type: "validateKnownField",
+        fieldName: field.field_name,
+        value: "error",
+      });
+      isValid = false;
+    }
+  }
+
+  for (const field of state.extraFields) {
+    if (field.required && !field.value) {
+      dispatch({
+        type: "validateExtraField",
+        fieldName: field.field_name,
+        value: "error",
+      });
+      isValid = false;
+    }
+  }
+
+  return isValid;
+};
+
+export const isSubmitDisabled = (state: FileUpload) => {
+  if (
```

Comment:
This logic can be reduced to just return the result of the expression and eliminating the if statement.

Reply:
Good call, removed this function entirely.